[WGSL] Remove _NO_ARGS version of RETURN_NODE macros

djg opened the pull request:
#### a9da78ab921397e0160caff7f1328b90e16f41d1
<pre>
[WGSL] Remove _NO_ARGS version of RETURN_NODE macros
<a href="https://bugs.webkit.org/show_bug.cgi?id=252233">https://bugs.webkit.org/show_bug.cgi?id=252233</a>
rdar://problem/105439637

Reviewed by Tadeu Zagallo.

By using __VA_OPT__ to selectively add a &apos;,&apos;, we can combine the two variants of
these macros into one.

* Source/WebGPU/WGSL/Parser.cpp:

Canonical link: <a href="https://commits.webkit.org/260278@main">https://commits.webkit.org/260278@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ba640c73622ce6945213dadfdad3b7d2a608986c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107670 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16722 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40566 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116818 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/116215 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/18144 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8040 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99843 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113424 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13696 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96893 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/41372 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95612 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28529 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/83211 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9705 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29878 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10382 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/6765 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15858 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49463 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7111 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/11942 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

tadeuzagallo commented on Source/WebGPU/WGSL/Parser.cpp:
```diff
@@ -46,26 +46,15 @@ namespace WGSL {
-#define RETURN_NODE_NO_ARGS(type) \
-    do { \
-        AST::type astNodeResult(CURRENT_SOURCE_SPAN()); \
+        AST::type astNodeResult(CURRENT_SOURCE_SPAN() __VA_OPT__(,) __VA_ARGS__); /* NOLINT */ \
```
Ha, TIL about `__VA_OPT__`. I think I used `##` somewhere, should probably remove that

tadeuzagallo reviewed (approved):
r=me

djg committed:
[WGSL] Remove _NO_ARGS version of RETURN_NODE macros
https://bugs.webkit.org/show_bug.cgi?id=252233
rdar://problem/105439637

Reviewed by Tadeu Zagallo.

By using __VA_OPT__ to selectively add a ',', we can combine the two variants of
these macros into one.

* Source/WebGPU/WGSL/Parser.cpp:

Canonical link: https://commits.webkit.org/260278@main
